app/src/main/java/com/github/libretube/helpers/ImportHelper.kt

```diff
@@ -144,47 +189,49 @@ object ImportHelper {
                     }
                     importPlaylists.add(playlist)
                 }
-            }
-            "application/json", "application/*", "application/octet-stream" -> {
-                val playlistFile = activity.contentResolver.openInputStream(uri)?.use {
-                    JsonHelper.json.decodeFromStream<ImportPlaylistFile>(it)
-                }
-                importPlaylists.addAll(playlistFile?.playlists.orEmpty())
-            }
-            else -> {
-                val message = activity.getString(R.string.unsupported_file_format, fileType)
-                activity.toastFromMainDispatcher(message)
-                return
-            }
-        }
 
-        // convert the YouTube URLs to videoIds
-        importPlaylists.forEach { playlist ->
-            playlist.videos = playlist.videos.map { it.takeLast(11) }
-        }
-        try {
-            PlaylistsHelper.importPlaylists(importPlaylists)
-            activity.toastFromMainDispatcher(R.string.success)
-        } catch (e: Exception) {
-            Log.e(TAG(), e.toString())
-            e.localizedMessage?.let {
-                activity.toastFromMainDispatcher(it)
+                // convert the YouTube URLs to videoIds
+                importPlaylists.forEach { importPlaylist ->
+                    importPlaylist.videos = importPlaylist.videos.map { it.takeLast(11) }
+                }
+                try {
+                    PlaylistsHelper.importNewPipePlaylists(importPlaylists)
+                    activity.toastFromMainDispatcher(R.string.success)
+                } catch (e: Exception) {
+                    Log.e(TAG(), e.toString())
+                    e.localizedMessage?.let {
+                        activity.toastFromMainDispatcher(it)
+                    }
+                }
```

Comment:
That try catch block is being called three times here (for each playlist type). Instead, it should only be there once after the `when` clause (which will work after applying my suggestion from `PlaylistsHelper.kt`.